test/MusicStore.E2ETests/PublishAndRunTestRunner.cs

```diff
@@ -1,81 +1,44 @@
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Logging.Testing;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace E2ETests
 {
-    [Trait("E2Etests", "PublishAndRun")]
-    public class PublishAndRunTests_X64 : LoggedTest
+    public class PublishAndRunTestRunner : LoggedTest
     {
-        public PublishAndRunTests_X64(ITestOutputHelper output) : base(output)
+        public PublishAndRunTestRunner(ITestOutputHelper output) 
+            : base(output)
         {
         }
 
-        [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux)]
-        [OSSkipCondition(OperatingSystems.MacOSX)]
-        public Task PublishAndRunTests_X64_WebListener_CoreCLR_Portable()
-        {
-            return RunTests(ServerType.WebListener, RuntimeFlavor.CoreClr, ApplicationType.Portable);
-        }
-
-        [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux)]
-        [OSSkipCondition(OperatingSystems.MacOSX)]
-        public Task PublishAndRunTests_X64_WebListener_CoreCLR_Standalone()
-        {
-            return RunTests(ServerType.WebListener, RuntimeFlavor.CoreClr, ApplicationType.Standalone);
-        }
-
-        [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux)]
-        [OSSkipCondition(OperatingSystems.MacOSX)]
-        public Task PublishAndRunTests_X64_WebListener_Clr()
-        {
-            return RunTests(ServerType.WebListener, RuntimeFlavor.Clr, ApplicationType.Portable);
-        }
-
-        [Fact]
-        public Task PublishAndRunTests_X64_Kestrel_CoreClr_Portable()
-        {
-            return RunTests(ServerType.Kestrel, RuntimeFlavor.CoreClr, ApplicationType.Portable);
-        }
-
-        [Fact]
-        public Task PublishAndRunTests_X64_Kestrel_CoreClr_Standalone()
-        {
-            return RunTests(ServerType.Kestrel, RuntimeFlavor.CoreClr, ApplicationType.Standalone);
-        }
-
-        [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux)]
-        [OSSkipCondition(OperatingSystems.MacOSX)]
-        public Task PublishAndRunTests_X64_Kestrel_Clr()
-        {
-            return RunTests(ServerType.Kestrel, RuntimeFlavor.Clr, ApplicationType.Standalone);
-        }
-
-        private async Task RunTests(
+        public async Task RunTests(
             ServerType serverType,
             RuntimeFlavor runtimeFlavor,
-            ApplicationType applicationType)
+            ApplicationType applicationType,
+            RuntimeArchitecture runtimeArchitecture)
         {
             var testName = $"PublishAndRunTests_{serverType}_{runtimeFlavor}_{applicationType}";
             using (StartLog(out var loggerFactory, testName))
             {
                 var logger = loggerFactory.CreateLogger("Publish_And_Run_Tests");
                 var musicStoreDbName = DbUtils.GetUniqueName();
 
+                // Specifying RuntimeArchitecture.x86 blows up regardless of RuntimeFlavor, but doesn't do anything
```

Comment:
I'll clear this up once https://github.com/aspnet/Hosting/issues/949 gets sorted.